pkg/sentry/socket/netfilter/tcp_matcher.go

```diff
@@ -142,10 +146,23 @@ func (tm *TCPMatcher) Match(hook stack.Hook, pkt stack.PacketBufferPtr, _, _ str
 
 	// Check whether the source and destination ports are within the
 	// matching range.
-	if sourcePort := tcpHeader.SourcePort(); sourcePort < tm.sourcePortStart || tm.sourcePortEnd < sourcePort {
+	// Take into account inverseFlags for DSTPT & SRCPT only
+	sPort := tcpHeader.SourcePort()
+	sPortMatch := sPort < tm.sourcePortStart || tm.sourcePortEnd < sPort
+	if tm.inverseFlags&linux.XT_TCP_INV_SRCPT == linux.XT_TCP_INV_SRCPT {
+		sPortMatch = !sPortMatch
+	}
```

Comment:
Instead of the `if` block, I think we can remove the branch via:
```
sPortMatch = sPortMatch != (tm.inverseFlags&linux.XT_TCP_INV_SRCPT == linux.XT_TCP_INV_SRCPT)
```

Same below for destination.

Reply:
I have made the suggested changes